src/clojure/nrepl/misc.clj

```diff
@@ -98,6 +99,14 @@
   [:ns :name :doc :file :arglists :forms :macro :special-form
    :protocol :line :column :added :deprecated :resource])
 
+(defn handle-file-meta [file]
+  (cond
```

Comment:
You can use `case` instead of `cond` and a `let` to remove most of the duplication.